machineconfiguration/v1alpha1/0000_80_machineconfignode-TechPreviewNoUpgrade.crd.yaml

```diff
@@ -0,0 +1,203 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: machineconfignodes.machineconfiguration.openshift.io
+  labels:
+    "openshift.io/operator-managed": ""
+  annotations:
+    api-approved.openshift.io: https://github.com/openshift/api/pull/1596
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
+    release.openshift.io/feature-set: TechPreviewNoUpgrade
+spec:
+  # group name to use for REST API: /apis/<group>/<version>
+  group: machineconfiguration.openshift.io
+  scope: Cluster
+  names:
+    kind: MachineConfigNode
+    singular: machineconfignode
+    plural: machineconfignodes
+    shortNames:
+      - mcnodes
+  versions:
+    - name: v1alpha1
+      # Each version can be enabled/disabled by Served flag.
+      served: true
+      # One and only one version must be marked as the storage version.
+      storage: true
+      subresources:
+        status: {}
+      additionalPrinterColumns:
+        - jsonPath: .status.conditions[?(@.type=="Updated")].status
+          name: Updated
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="UpdatePreparing")].status
+          name: UpdatePreparing
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="UpdateInProgress")].status
+          name: UpdateInProgress
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="UpdatePostAction")].status
+          name: UpdatePostAction
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="UpdateCompleting")].status
+          name: UpdateCompleting
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="Resuming")].status
+          name: Resuming
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="ComparingMCs")].status
+          name: ComparingMachineConfigs
+          priority: 1
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="ApplyingFilesAndOS")].status
+          name: UpdatingFilesAndOS
+          priority: 1
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="DrainingNode")].status
+          name: DrainingNode
+          priority: 1
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="CordoningNode")].status
+          name: CordoningNode
+          priority: 1
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="RebootingNode")].status
+          name: RebootingNode
+          priority: 1
+          type: string
+        - jsonPath: .status.conditions[?(@.type=="ReloadingCRIO")].status
+          name: ReloadingCRIO
+          priority: 1
```

Comment:
yes. Some flip flop like reloading/rebooting since they are mutually exclusive upgrade events but otherwise I ordered these so it would look like a cascade.

Reply:
Ok, I'm curious then, when does cordoning happen after draining? My understanding is that you cordon a node so that you can drain it, is there some additional cordoning happening as well?